mcp_server/__main__.py

```
@@ -0,0 +1,67 @@
+            output_schema=create_model(output_schema),
+            browser=ctx.request_context.lifespan_context.browser,
+        )
+
+        if source is None:
+            await ctx.warning("Analysis completed but no relevant data found")
+            return "Analysis failed: No relevant data found"
+
+        await ctx.info("Analysis completed successfully")
+    except Exception as e:
+        await ctx.error(f"Error during analysis: {e!s}")
+        raise Exception(f"Error during analysis: {e!s}") from e
```

Ruff (0.12.2)

Create your own exception

(TRY002)